src/main/java/com/ericsson/ei/erqueryservice/ERQueryService.java

```diff
@@ -60,97 +61,58 @@ public void setHttpRequest(HttpRequest request) {
-            } else {
-                LOGGER.info("The URL to ER is not provided");
-            }
-        } catch (MalformedURLException e) {
-            LOGGER.error("Error while building the ER url. Stacktrace: {}", ExceptionUtils.getStackTrace(e));
-        } catch (Exception e) {
-            LOGGER.error("Error occurred while executing REST GET to: {} for {}", erUrl, eventId, e);
+            ResponseEntity requestToER = sendRequestToER(eventId, searchOption, limit, levels,
+                    tree);
+            return requestToER;
+        } catch (IOException | URISyntaxException e) {
+            throw new Exception("Error occurred while executing REST POST", e);
         }
-
-        return null;
     }
 
-    /**
-     * This method is used to fetch only the upstream or downstream or both
-     * event information from ER2.0 based on the eventID and searchOption
-     * conditions.
-     *
-     * @param eventId
-     *            the id of the event.
-     * @param searchOption
-     *            the SearchOption to indicate whether to search up, down or
-     *            both ways from the eventId.
-     * @param limit
-     *            sets the limit of how many events up and/or down stream from
-     *            the eventId to include in the result.
-     * @param levels
-     *            sets the limit of how many levels up and/or down stream from
-     *            the eventId to include in the result.
-     * @param tree
-     *            whether or not to retain the tree structure in the result.
-     * @return ResponseEntity
-     */
-    public ResponseEntity getEventStreamDataById(String eventId, SearchOption searchOption, int limit,
-            int levels, boolean tree) {
-
-        String uri = null;
-        try {
-            if(StringUtils.isNotBlank(erBaseUrl)) {
-                // Request Body parameters
-                final SearchParameters searchParameters = getSearchParameters(searchOption);
-                request
-                        .setHttpMethod(HttpMethod.POST)
-                        .setBaseUrl(erBaseUrl)
-                        .setEndpoint(eventId)
-                        .addParam("limit", Integer.toString(limit))
-                        .addParam("levels", Integer.toString(levels))
-                        .addParam("tree", Boolean.toString(tree))
-                        .setBody(searchParameters.getAsJsonString(), ContentType.APPLICATION_JSON);
-
-                uri = request.getURI().toString();
-                LOGGER.debug("The URL to ER is: {}", uri);
-
-                return request.performRequest();
-            } else {
-                LOGGER.info("The URL to ER is not provided");
-            }
-        }
-        catch (Exception e) {
-            LOGGER.error("Error occurred while executing REST POST to {}, stacktrace: {}", uri, e);
+    private ResponseEntity sendRequestToER(String eventId, SearchOption searchOption, int limit,
```

Comment:
I would not mind splitting this into 2 methods, one for preparing the httpRequest, and one for executing the httpRequest. Now those 2 different logical operations is done in the same method.

Reply:
Done

Reply:
See comment @m-linner-ericsson comment below